tachyon/zk/plonk/circuit/floor_planner/lookup_table_assigner.h

```diff
@@ -0,0 +1,95 @@
+#ifndef TACHYON_ZK_PLONK_CIRCUIT_FLOOR_PLANNER_LOOKUP_TABLE_ASSIGNER_H_
+#define TACHYON_ZK_PLONK_CIRCUIT_FLOOR_PLANNER_LOOKUP_TABLE_ASSIGNER_H_
+
+#include <stddef.h>
+
+#include <optional>
+#include <utility>
+#include <vector>
+
+#include "absl/container/flat_hash_map.h"
+
+#include "tachyon/base/functional/identity.h"
+#include "tachyon/base/logging.h"
+#include "tachyon/zk/plonk/circuit/assignment.h"
+#include "tachyon/zk/plonk/circuit/floor_planner/scoped_region.h"
+#include "tachyon/zk/plonk/circuit/floor_planner/simple_lookup_table_layouter.h"
+#include "tachyon/zk/plonk/circuit/layouter.h"
+#include "tachyon/zk/plonk/circuit/lookup_table.h"
+#include "tachyon/zk/plonk/circuit/lookup_table_column.h"
+
+namespace tachyon::zk {
+
+template <typename F>
+class LookupTableAssigner {
+ public:
+  using AssignLookupTableCallback =
+      typename Layouter<F>::AssignLookupTableCallback;
+
+  LookupTableAssigner(Assignment<F>* assignment,
+                      std::vector<LookupTableColumn>& lookup_table_columns)
```

Comment:
```suggestion
                      const std::vector<LookupTableColumn>& lookup_table_columns)
```

Reply:
This shouldn't be a const cuz it pushes values into it.